src/python/pants/backend/python/macros/pants_requirement_caof.py

```diff
@@ -45,6 +46,27 @@ def __call__(
         :param modules: The modules exposed by the dist, e.g. `['pants.testutil']`. This defaults
             to the name of the dist without the leading `pantsbuild`.
         """
+        warn_or_error(
+            "2.10.0.dev0",
+            "the `pants_requirement` macro",
+            (
+                "Use the target generator `pants_requirements` instead. Specifically, we recommend "
```

Comment:
I think it is irrelevant to mention "target generator" here.. and may add confusion.